Sources/Widgets/Widgets3D/ShapeWidget/example/index.js

```diff
@@ -276,10 +277,62 @@ reader
       }
     );
 
+    const updateWidgetVisibility = (position) => {
+      for (let i = 0; i < 3; i++) {
+        if (position[i] !== 0) {
+          /* testing if the widget is on the slice and has been placed to modify visibility */
+          const rectangleWidgetVisibility =
+            widgets.rectangleWidget
+              .getWidgetState()
+              .getPoint1Handle()
+              .getOrigin()[i] === position[i] ||
+            !scene.widgetManager.getWidgets()[0].getPoint1();
+          const ellipseWidgetVisibility =
+            widgets.ellipseWidget
+              .getWidgetState()
+              .getPoint1Handle()
+              .getOrigin()[i] === position[i] ||
+            !scene.widgetManager.getWidgets()[1].getPoint1();
+          const circleWidgetVisibility =
+            widgets.circleWidget.getWidgetState().getPoint1Handle().getOrigin()[
+              i
+            ] === position[i] ||
+            !scene.widgetManager.getWidgets()[2].getPoint1();
+          widgets.rectangleWidget.setVisibility(rectangleWidgetVisibility);
+          widgets.ellipseWidget.setVisibility(ellipseWidgetVisibility);
+          widgets.circleWidget.setVisibility(circleWidgetVisibility);
+        }
+      }
+    };
+
     const update = () => {
       const slicingMode = image.imageMapper.getSlicingMode() % 3;
 
+      scene.renderWindow.render();
```

Comment:
does not seem right to refresh the render window before updating the widgets

Reply:
it is updating the slice, it allows the UI to refresh when the user drags the scroll bar and prevents a refresh when the user moves mouse on the scene after scrolling the bar

Reply:
can't it be done at the end of the update() function ?